tests/net/dhcpv4/src/main.c

```diff
@@ -156,14 +156,12 @@ struct dhcp_msg {
 
 static struct k_sem test_lock;
 
-static void test_result(bool pass)
-{
-	if (pass) {
-		TC_END_REPORT(TC_PASS);
-	} else {
-		TC_END_REPORT(TC_FAIL);
-	}
-}
+/* As per subsys/net/ip/dhcpv4.c
+ * "RFC2131 4.1.1 requires we wait a random period between 1 and 10
+ * seconds before sending the initial discover". So test will wait
+ * max 11 seconds.
```

Comment:
What? Test waits up to 11 seconds? That's too long. If even small percentage of tests will do tricks like that, the testsuite will take enormous time to run (and we have regulaa CI timeouts already). Please limit that to 2s, max 3s. If that requires delay in the stack code to be configurable, well, do that, and override it in test config.

Reply:
https://github.com/rveerama1/zephyr/blob/master/subsys/net/ip/dhcpv4.c#L1056

Reply:
Otherwise, I have to introduce new config option for DHCPv4 testing and make it 0 to 2s max.

Reply:
> Otherwise, I have to introduce new config option for DHCPv4 testing and make it 0 to 2s max

Yes, that's what I suggest, make DHCPV4_INITIAL_DELAY_MAX into CONFIG_DHCPV4_INITIAL_DELAY_MAX. Because today I ran some POSIX test, and it also ran noticeable number of seconds. If we let those expand uncontrollably, it will be havoc. Also ping @nashif, @galak for suggestions.

Reply:
Proposed some changes, please take a look.

Reply:
> Proposed some changes, please take a look.

Looks good. I'd personally just make that a normal option to cut on the number of if's, after all, if someone will want to break standard, they'll do that with or without option.

But you introduced the option, but not set in the test config, so test fails.

Reply:
depends on NET_DHCPV4
	depends on ZTEST
	default 2
Isn't it enabled by default if NET_DHCPV4 and ZTEST are enabled.
Without specifying in test prj.conf I can see it in generated .config file.
CI fails, I am investigating it.

Reply:
> Looks good. I'd personally just make that a normal option to cut on the number of if's, after all, if someone will want to break standard, they'll do that with or without option.

Ok, I will try that.